easypermissions/src/main/java/pub/devrel/easypermissions/RationaleAppCompatDialogFragment.java

```diff
@@ -0,0 +1,114 @@
+package pub.devrel.easypermissions;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+
+import java.util.Arrays;
+
+import static pub.devrel.easypermissions.EasyPermissions.executePermissionsRequest;
+
+
+public class RationaleAppCompatDialogFragment extends DialogFragment implements Dialog.OnClickListener {
+
+    private EasyPermissions.PermissionCallbacks permissionCallbacks;
+
+    private int positiveButton;
+    private int negativeButton;
+    private int requestCode;
+    private String rationaleMsg;
+    private String[] permissions;
+
+    public static RationaleAppCompatDialogFragment newInstance(@StringRes int positiveButton,
+                                                               @StringRes int negativeButton,
+                                                               @NonNull String rationaleMsg,
+                                                               int requestCode,
+                                                               @NonNull String[] permissions) {
+        RationaleAppCompatDialogFragment dialogFragment = new RationaleAppCompatDialogFragment();
+        Bundle bundle = new Bundle();
+        bundle.putInt("positiveButton", positiveButton);
```

Comment:
i don't think having them static is a good idea tbh, we don't want to hold on variables and do the hard work of GC them. having them as a parameter is a good practice.